### PR TITLE
Update faq.md

### PR DESCRIPTION
**Pull request by mit-mit**
Fixes https://github.com/flutter/flutter/issues/9806

### CODE REVIEW

File: faq.md [hunk 1]
```diff
@@ -281,6 +281,17 @@ is retained after a reload. This means you can quickly iterate
 on a screen deeply nested in your app, without starting
 from the home screen after every reload.
 
+### How is 'hot reload' different from 'full restart'?
```
Comment (sethladd) [inline faq.md]: Do we also link to https://github.com/flutter/flutter/wiki/Hot-Reload or delete https://github.com/flutter/flutter/wiki/Hot-Reload ? (or, merge https://github.com/flutter/flutter/wiki/Hot-Reload into here?)

Comment (mit-mit) [inline faq.md]: I'll delete that Wiki page, it's not up-to-date.

Comment (sethladd) [inline faq.md]: Great, thanks!

**Comment by sethladd** [comment]
lgtm

File: faq.md (hunk 1 above)
Comment (mit-mit) [inline faq.md]: Sounds good

Comment (mit-mit) [inline faq.md]: Sounds good

**Pull request merged**